api-client/src/endpoints/hooks/useAllPossibleSearchResults.ts

```diff
@@ -0,0 +1,98 @@
+import { useCallback, useMemo } from 'react'
+
+import createCategoriesEndpoint from 'api-client/src/endpoints/createCategoriesEndpoint'
+import createEventsEndpoint from 'api-client/src/endpoints/createEventsEndpoint'
+import useLoadAsync from 'api-client/src/endpoints/hooks/useLoadAsync'
+import useLoadFromEndpoint from 'api-client/src/endpoints/hooks/useLoadFromEndpoint'
+import loadSprungbrettJobs from 'api-client/src/endpoints/loadSprungbrettJobs'
+
+type useSearchParams = {
+  city: string
+  language: string
+  cmsApiBaseUrl: string
+}
+
+export type SearchResult = {
```

Comment:
I'm wondering if it makes sense to add the component (that will be rendered) to the search result. This could slightly speed up searching, as we are rendering all items initially anyways.

Reply:
It might also delay the start of the search because it would take longer to load everything in, and that is currently by far the bottleneck :(

Reply:
But everything is loaded anyway, when opening the search, right?

Reply:
Yes, everything loads then, but an array of components is a lot bigger than an array of tiny objects, so it would take longer to add it to the Minisearch. And since that is already the moment when the most waiting happens (because we're waiting for the categories etc to load), I'd rather not make it even longer, even though it would marginally speed up the display once the search is actually underway. Which feels pretty fast to me already

Reply:
Fine. Then, let's not do this. :)